firebase.go

```diff
@@ -146,11 +151,7 @@ func (fb *Firebase) copy() *Firebase {
 }
 
 func (fb *Firebase) makeRequest(method string, body []byte) (*http.Request, error) {
-	path := fb.url + "/.json"
-
-	if len(fb.params) > 0 {
-		path += "?" + fb.params.Encode()
-	}
+	path := fb.String()
```

Comment:
style nit: this declaration is small enough that we might inline it below and save the allocation of the `path` var.

```go
return http.NewRequest(method, fb.String(), bytes.NewReader(body))
```